src/dinitctl.cc

```diff
@@ -324,8 +359,52 @@ int dinitctl_main(int argc, char **argv)
             cmdline_error = true;
         }
     }
+    else if (command == command_t::SIG_SEND) {
+        if (show_siglist) {
+            if (sigstr.empty()) {
+                command = command_t::SIG_LIST;
+            }
+            else {
+                cmdline_error = true;
+            }
+        }
+        else {
+            if (sigstr.empty()) {
+                cerr << "dinitctl: signal number/name must be specified" << std::endl;
+                return 1;
+            }
+            else if (sigstr == "NONE" | sigstr == "none") {
+                cerr << "dinitctl: '" << sigstr
+                    << "' is not a valid signal name/number" << std::endl;
+                return 1;
+            }
+            if (service_name == nullptr) {
+                cerr << "dinitctl: service name must be specified" << std::endl;
+                return 1;
+            }
+            sig_num = dinit_load::signal_name_to_number(sigstr);
+            if (sig_num == -1) {
+                try
+                {
+                    size_t pos;
+                    sig_num = std::stoi(sigstr, &pos);
+                    if (sigstr.size() != pos) {
+                        throw std::invalid_argument("");
+                    }
+                }
+                catch (std::exception)
```

Comment:
Ok, but at least add a comment saying what exception types this is expected to handle.

Someone reading this code should be able to look at it and think "Ok, this is designed to handle both invalid_argument and out_of_range".